resources: add packer files to make x86 and arm npb image

Harshil2107 opened the pull request:
- This PR renames the `npb` directory to `npb-24.04-imgs`
- Adds packer files to make the x86 and arm npb images
- Adds other necessary files like `post-install.sh` that are required by the image to build the NPB benchmarks

Harshil2107 committed:
resources: add packer files to make x86 and arm npb image

erin-le commented on src/x86-ubuntu/scripts/post-installation.sh:
Is this file needed?

Harshil2107 replied:
reverted the changes made to the file

Harshil2107 committed:
revert changes for x86-ubuntu directory

Harshil2107 committed:
resources: rename files and update documentation and gitignore

Harshil2107 committed:
resources: Update sha checksum for x86

Harshil2107 committed:
resources: update checksum

Harshil2107 committed:
resources: Add section about creating flash0.img

Harshil2107 committed:
resources: Added commands to build flash0.img in build-arm.sh

erin-le commented on src/npb-24.04-imgs/build-arm.sh:
```diff
@@ -0,0 +1,23 @@
+fi
+
+# make the flash0.sh file
+cd ./files
```
```suggestion
mkdir files
cd ./files
```

erin-le commented on src/npb-24.04-imgs/arm-npb.pkr.hcl:
```diff
@@ -0,0 +1,76 @@
+                        ["-enable-kvm"],
+                        ["-machine", "virt"],
+                        ["-machine", "gic-version=3"],
+                        ["-vga", "virtio"],
```
```suggestion
```

erin-le committed:
resources: modify arm build scripts to correct errors

This commit adds to build-arm.sh so the commands to get flash0.img work and also removes arguments from arm-npb.pkr.hcl so that the image will build.

erin-le commented on src/npb-24.04-imgs/README.md:
```diff
@@ -0,0 +1,186 @@
+
+The disk has network disabled by default to improve boot time in gem5.
+
+If you want to enable networking, you need to modify the disk image and move the file `/etc/netplan/50-cloud-init.yaml.bak` to `/etc/netplan/50-cloud-init.yaml.bak`.
```
These appear to be the same file path.

erin-le commented on src/npb-24.04-imgs/README.md:
```diff
@@ -0,0 +1,186 @@
+to run the arm disk image.
+After this process succeeds, the disk image can be found on the `npb-24.04-imgs/disk-image-x86-npb/disk-image-x86-npb` or `npb-24.04-imgs/disk-image-arm-npb/disk-image-arm-npb` repectively.
+
+This gapbs image uses the prebuilt ubuntu 24.04 image as a base image. The gapbs image also throws the same exit events as the base image.
```
```suggestion
This npb image uses the prebuilt ubuntu 24.04 image as a base image. The npb image also throws the same exit events as the base image.
```

Harshil2107 committed:
Update src/npb-24.04-imgs/README.md

Co-authored-by: Erin (Jianghua) Le <jianghuale@gmail.com>

Harshil2107 committed:
resources: Update readme

erin-le reviewed (approved):
LGTM. I tested building both disk images and used them as resources to run UA size s. I had some trouble writing/getting the config scripts to work, so it could be a good idea to add  example scripts for these disk images in configs/example/gem5_library, then update the documentation to point to them.
